sources/academy/webscraping/scraping_basics_python/05_parsing_html.md

```diff
@@ -0,0 +1,211 @@
+---
+title: Parsing HTML with Python
+sidebar_label: Parsing HTML
+description: Lesson about building a Python application for watching prices. Using the Beautiful Soup library to parse HTML code of a product listing page.
+sidebar_position: 5
+slug: /scraping-basics-python/parsing-html
+---
+
+**In this lesson we'll look for products in the downloaded HTML. We'll use BeautifulSoup to turn the HTML into objects which we can work with in our Python program.**
+
+---
+
+From lessons about browser DevTools we know that the HTML tags representing individual products have a `class` attribute which, among other values, contains `product-item`.
+
+![Products have the ‘product-item’ class](./images/product-item.png)
+
+As a first step, let's try counting how many products are on the listing page.
+
+## Treating HTML as a string
+
+Currently, the entire HTML is available in our program as a string. Our program can print it to the screen or save it to a file, but not much more. If it's a string, could we use Python string operations to count the products? Each Python string has `.count()`, a [method for counting substrings](https://docs.python.org/3/library/stdtypes.html#str.count).
```

Comment:
we could even make the `.count` a link might be better than leaving it at later stage. 

```suggestion
Currently, the entire HTML is available in our program as a string. Our program can print it to the screen or save it to a file, but not much more. If it's a string, could we use Python string operations to count the products? Each Python string has <code>[.count()](https://docs.python.org/3/library/stdtypes.html#str.count)</code>, a method for counting substrings.
```

Reply:
This works too:
```markdown
[`.count()`](https://example.com)
```
I tried to avoid links mixed with code, as my first impression was that with the current CSS it's not obvious they're clickable. But I gave it a second look now and I don't feel the same way 🤷‍♂️ I think I'll use it—at more places than just here.

Reply:
Addressed in 43681351